### ISSUE TITLE
handle duplicate logins in registration flow

### ISSUE THREAD

**Issue by dawba** [dawba/market-place#26]
During testing the email verification system I have encountered a gap in our User handling.

When registering a new account with new email but existing login in our database, the app throws an error and it is not handled gracefully. The HTTP response contains nothing, thus making potential frontend of our application not know what really happened. We need to add error handling here.

Steps to recreate:
1. Create user1 `api/user/register`
```
{
    "login": "login",
    "password": "password",
    "email":"email1@email.com",
    "phoneNumber": "518274407"
}
```
Observe: successful account creation.

2. Create user2
```
{
    "login": "login",
    "password": "password",
    "email":"email2@email.com",
    "phoneNumber": "518274407"
}
```

Observe: app throws an error, we get empty response body.
Error thrown:
`java.sql.SQLIntegrityConstraintViolationException: Duplicate entry 'login' for key 'user.UK_ew1hvam8uwaknuaellwhqchhb'`

**Comment by Konrad1w** [comment]
I have done it, you can check it in #28 PR

**Comment by dawba** [comment]
Resolved by #28
